docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md

```diff
@@ -0,0 +1,815 @@
+        kind: Role
+        name: {{RESOURCE_NAME}}
+      subjects:
+      - kind: ServiceAccount
+        name: {{RESOURCE_NAME}}
+# This will build the container image before deployment
+- name: outerloop-build
+  image:
+    dockerfile:
+      buildContext: ${PROJECT_SOURCE}
+      rootRequired: false
+      uri: ./Dockerfile
+    imageName: "{{CONTAINER_IMAGE}}"
+# This will create a Deployment in order to run your container image across the cluster.
+# Note that we expose the env vars necessary to connect application with the mongodb service.
+- name: outerloop-deployment
+  kubernetes:
+    inlined: |
+      kind: Deployment
+      apiVersion: apps/v1
+      metadata:
+        name: {{RESOURCE_NAME}}
+      spec:
+        replicas: 1
+        selector:
+          matchLabels:
+            app: {{RESOURCE_NAME}}
+        template:
+          metadata:
+            labels:
+              app: {{RESOURCE_NAME}}
+          spec:
+            containers:
+              - name: {{RESOURCE_NAME}}
+ image: {{CONTAINER_IMAGE}}
+ ports:
+   - name: http
+ containerPort: {{CONTAINER_PORT}}
+ protocol: TCP
+ env:
+ - name: username
+   value: {{USERNAME}}
+ - name: host
+   value: {{HOST}}
+ - name: password
+   valueFrom:
+ secretKeyRef:
+ name: mongodb
+ key: mongodb-root-password
+ resources:
+   limits:
+ memory: "1024Mi"
+ cpu: "500m"
+# This will create a Service so your Deployment is accessible.
+# Depending on your cluster, you may modify this code so it's a
+# NodePort, ClusterIP or a LoadBalancer service.
+- name: outerloop-service
+  kubernetes:
+    inlined: |
+      apiVersion: v1
+      kind: Service
+      metadata:
+        name: {{RESOURCE_NAME}}
+      spec:
+        ports:
+        - name: "{{CONTAINER_PORT}}"
+          port: {{CONTAINER_PORT}}
+          protocol: TCP
+          targetPort: {{CONTAINER_PORT}}
+        selector:
+          app: {{RESOURCE_NAME}}
+        type: NodePort
+- name: outerloop-url
+  kubernetes:
+    inlined: |
+      apiVersion: networking.k8s.io/v1
+      kind: Ingress
+      metadata:
+        name: {{RESOURCE_NAME}}
+      spec:
+        rules:
+          - host: "{{DOMAIN_NAME}}"
+            http:
+              paths:
+ - path: "/"
+   pathType: Prefix
+   backend:
+ service:
+ name: {{RESOURCE_NAME}}
+ port:
+   number: {{CONTAINER_PORT}}
+```
+
+### Add the `variables`
+Next, we add a `variables` section to the Devfile, so that we can make use of the same variables at multiple locations within the Devfile.
+
+```yaml
+# Add the following variables code anywhere in devfile.yaml
+# This MUST be a container registry you are able to access
+variables:
+  CONTAINER_IMAGE: quay.io/MYUSERNAME/go-odo-example
+  RESOURCE_NAME: my-go-app
+  CONTAINER_PORT: "8080"
+  DOMAIN_NAME: go.example.com
+  USERNAME: root
+  HOST: mongodb
+```
+
+:::note
+Ensure that you replace `MYUSERNAME` in `CONTAINER_IMAGE` with your own username; or use a container registry that you have the write permissions to.
```

Comment:
Without changing the Devfile, can you leverage variable substitution instead? By running `odo deploy --var CONTAINER_IMAGE=...`, I think it makes things easy to manage, even from the user perspective, if they want to push to a completely different registry, not just `quay.io`..

Reply:
Oh, you're right, that is a good idea.